_docs/developer/getting_started/vm_install_using_vagrant.md

```diff
@@ -46,212 +52,475 @@ If you using an Intel-based Mac, you will follow the instructions below._
+   >
+   > 7. Enter your **BIOS** (generally by pressing Del, F12, or other keys
+   > while booting). If you are not able to find the key combo needed to enter
+   > your BIOS, refer to [this guide](https://www.tomshardware.com/reviews/bios-keys-to-access-your-firmware,5732.html).
+   >
+   > 8. Locate **Virtualization**, and enable it. (Note: If you cannot find the
+   > option to enable virtualization, [search Google](http://tinyurl.com/enable-virtualization)
+   > for a tutorial on enabling it with your motherboard.)
+   >
+   > 9. Reboot your computer.
+
+   > **Ubuntu**
+   > 1. Enter your **BIOS** (generally by pressing Del, F12, or other keys
+   > while booting).
+   >
+   > 2. Navigate the **BIOS Settings**.
+   >
+   > 3. Locate **Virtualization** and enable it.
+   >
+   > 4. Be sure to choose **Hardware Virtualization** in the **System ->
+   > Acceleration** settings of the virtual machine you are using.
+   >
+   >  **NOTE** 
+   >  If using secure boot, vagrant may fail to work with VirtualBox. You will
+   >  then either need to disable secure boot from the boot menu/BIOS or follow 
+   > [these steps](https://era86.github.io/2018/01/24/vagrant-virtualbox-secureboot-in-ubuntu-1604.html)
+   > to self-sign the necessary packages to run vagrant and VirtualBox.
+
+   <br>
 
-   4. Once your PC has rebooted, click the **Troubleshoot** option.
+2. Download and install the latest version of [Ruby](https://www.ruby-lang.org/en/downloads).
+   
+   <br>
 
-   5. Click **Advanced Options**.
+3. Download and install the latest version of [Git](https://git-scm.com/downloads).
+   
+   <br>
+
+4. Download and install [VirtualBox 6](https://www.virtualbox.org/wiki/Download_Old_Builds_6_1)
+   and [Vagrant](https://www.vagrantup.com). Virtual Box 7 may not work
+   currently with Submitty:
+
+   > **Windows 10**
+   >
+   > * You can just go to the respective sites and download the necessary
+   > binaries.
+
+   >**MacOS**
+   >
+   > * You can either go to respective sites and download the necessary binaries
+   > or install [homebrew](http://brew.sh/) if you don't have it and then run:
+   >
+   >     ```
+   >     brew install --cask virtualbox
+   >     brew install --cask vagrant
+   >     ```
+
+   > **Ubuntu/Debian**
+   >
+   > * The Ubuntu repository does not contain the latest version of Vagrant or
+   > VirtualBox and using them may not work nor are they supported. We recommend
+   > that you either download the necessary binaries from their respective steps
+   > or follow the steps outlined below for each:
+   >
+   >    VirtualBox: <https://www.virtualbox.org/wiki/Linux_Downloads>
+   >    
+   >    Vagrant: <https://developer.hashicorp.com/vagrant/downloads> 
+   >    (if that doesn't work, try: <https://vagrant-deb.linestarve.com/>)
+   
+   <br>
+   
+5. Clone [the Submitty repository](https://github.com/Submitty/Submitty) to a
+   location on your computer (the "host").
```

Comment:
I'd suggest adding a note here saying something like:

the location must be a normal file directory, do not use an synced directory like dropbox/onedrive etc since this typically causes permissions issues / undefined behavior 

^ which seems to be a common problem on windows new developers run into